src/components/Table/TableBase.tsx

```diff
@@ -377,12 +380,17 @@ export class TableBase<T extends {}> extends React.Component<
 			newState.checkedItems !== this.state.checkedItems;
 		this.setState(newState);
 		if (this.props.onCheck && hasCheckedItemsChanged) {
-			this.props.onCheck(newState.checkedItems);
+			this.props.onCheck(newState.checkedItems, this.state.allChecked);
 		}
 	};
 
+	private toggleCheckedState = () => {
+		const { allChecked } = this.state;
+		return allChecked === 'none' ? 'all' : 'none';
```

Comment:
Would it make sense if this was v ?
```suggestion
		return allChecked === 'all' ? 'none' : 'all';
```
This way:
none -> all
some -> all
all -> none

Reply:
that's not what we have today, should we change it to this version ? @thgreasi

Reply:
I hadn't realized this. Ignore this, let's preserve the current behavior.